### PR TITLE
datalibs: introduce folder management; fix various glitches; refactor

### LINKED ISSUE #1570: display 'message' field in data libraries
that disappeared in the transition from old to new UI

related to https://github.com/galaxyproject/galaxy/issues/1177

reported by Devon Ryan: https://biostar.usegalaxy.org/p/15755/

### PR DESCRIPTION
**Pull request by martenson**
ping @blankenberg 
fixes #1570

### CODE REVIEW

**Comment by martenson** [comment]
xref https://github.com/galaxyproject/galaxy/issues/1177

**Comment by martenson** [comment]
rebased due to conflicts, asking for a review

File: lib/galaxy/webapps/galaxy/api/folder_contents.py [hunk 1]
```diff
@@ -83,7 +83,9 @@ def index( self, trans, folder_id, **kwd ):
                 encoded_id = 'F' + encoded_id
                 can_modify = is_admin or ( trans.user and trans.app.security_agent.can_modify_library_item( current_user_roles, folder ) )
                 can_manage = is_admin or ( trans.user and trans.app.security_agent.can_manage_library_item( current_user_roles, folder ) )
-                return_item.update( dict( can_modify=can_modify, can_manage=can_manage ) )
+                return_item.update( dict( can_modify=can_modify, can_manage=can_manage, description=content_item.description ) )
```
Comment (nsoranzo) [inline lib/galaxy/webapps/galaxy/api/folder_contents.py]: The above change seems not needed, it's done better below.

Comment (martenson) [inline lib/galaxy/webapps/galaxy/api/folder_contents.py]: fixed, thank you

**Comment by carlfeberhard** [comment]
I'm not Dan, but it seems to work in my limited testing: +1

'Message' has become 'description', then it looks like?

Let me know if you want a merge.

**Comment by martenson** [comment]
@carlfeberhard `message` is how it used to be called in old libs, but in fact the usage is better described by 'description' (which is also consistent with what it is called for libraries, folders and in other places). Did not figure out a better way to unify it. :/

**Comment by carlfeberhard** [comment]
@martenson Yeah - no worries - I think that's fine and it's your show. I just wanted to make sure I understood the change and where to look to test it.

**Comment by blankenberg** [comment]
Could change message/description to "Notes" or "Comments" (I don't have strong feelings about any particular naming).

**Comment by martenson** [comment]
@blankenberg  we use 'description' all over the place, I would rather stick with it

**Comment by martenson** [comment]
failing API test is unrelated, see https://github.com/galaxyproject/galaxy/issues/1617

**Comment by martenson** [comment]
please merge this, downstream work is waiting for this

**Pull request merged**